drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_trigger.c

```diff
@@ -254,33 +284,50 @@ int lsm6dsv16x_init_interrupt(const struct device *dev)
 	lsm6dsv16x->work.handler = lsm6dsv16x_work_cb;
 #endif /* CONFIG_LSM6DSV16X_TRIGGER_OWN_THREAD */
 
-	ret = gpio_pin_configure_dt(lsm6dsv16x->drdy_gpio, GPIO_INPUT);
-	if (ret < 0) {
-		LOG_DBG("Could not configure gpio");
-		return ret;
-	}
+	if (!ON_I3C_BUS(cfg)) {
+		ret = gpio_pin_configure_dt(lsm6dsv16x->drdy_gpio, GPIO_INPUT);
+		if (ret < 0) {
+			LOG_DBG("Could not configure gpio");
+			return ret;
+		}
 
-	gpio_init_callback(&lsm6dsv16x->gpio_cb,
-			   lsm6dsv16x_gpio_callback,
-			   BIT(lsm6dsv16x->drdy_gpio->pin));
+		gpio_init_callback(&lsm6dsv16x->gpio_cb,
+				lsm6dsv16x_gpio_callback,
+				BIT(lsm6dsv16x->drdy_gpio->pin));
```

Comment:
be careful to indentation here. Maybe you may want to use uncrustify to help on this side.

Reply:
I pushed a new commit after running uncrustify. It just fix few indentation issues. Pls squash with proper commit after evaluating it

Reply:
done, thanks